src/features_creator/features_creator.py

```diff
@@ -192,4 +196,19 @@ def calculate_average(df, pattern):
         2              3.0
 
     """
-    ...
+    # check input type
+    if not isinstance(df, pd.DataFrame):
+        raise TypeError("df must be a pandas dataframe")
+    if not isinstance(pattern, str):
+        raise TypeError("pattern must be a string")
+
+    # get matching columns
+    columns = get_matching_column_names(df, pattern)
+
+    # calculate average from matching columns
+    df_avg = df[columns].mean(axis=1)
+
+    # convert to np array
+    df_avg = df_avg.to_numpy()
```

Comment:
Since here, you decide to return a NumPy array, maybe also need to change the example in docstring.

Reply:
good catch!